stacks-signer/src/runloop.rs

```diff
@@ -788,15 +861,73 @@ impl<C: Coordinator> RunLoop<C> {
             "Sending {} messages to other stacker-db instances.",
             outbound_messages.len()
         );
+        let coordinator_info = calculate_coordinator(
+            &self.signing_round.public_keys,
+            &self.stacks_client,
+            &self.nack_messages_received,
+            &self.nack_threshold,
+        );
         for msg in outbound_messages {
-            let ack = self.stackerdb.send_message_with_retry(msg.into());
+            let packet_metadata =
+                wrap_packet_in_packet_metadata(msg, coordinator_info.coordinator_metadata.clone());
+            let ack = self
+                .stackerdb
+                .send_message_with_retry(packet_metadata.into());
             if let Ok(ack) = ack {
                 debug!("ACK: {:?}", ack);
             } else {
                 warn!("Failed to send message to stacker-db instance: {:?}", ack);
             }
         }
     }
+    fn process_signer_consensus_hash_view(
+        &mut self,
+        current_signer_coordinator_metadata: &CoordinatorMetadata,
+        packet_sender_coordinator_metadata: &CoordinatorMetadata,
+    ) {
+        let current_signer_consensus_hash =
+            current_signer_coordinator_metadata.stacks_consensus_hash;
+        let current_signer_block_height = current_signer_coordinator_metadata.stacks_block_height;
+        let packet_sender_consensus_hash = packet_sender_coordinator_metadata.stacks_consensus_hash;
+        let packet_sender_block_height = packet_sender_coordinator_metadata.stacks_block_height;
+        if current_signer_consensus_hash != packet_sender_consensus_hash {
+            if current_signer_block_height < packet_sender_block_height {
+                warn!(
+                    "Signer {} has an outdated view of stacks chain",
+                    self.signing_round.signer_id
+                );
+            } else {
+                self.broadcast_nack_message(current_signer_coordinator_metadata.clone());
+            }
+        }
```

Comment:
I'm not exactly sure what the spec is here, but my assumption would be that the signer should broadcast a `nack` for any messages with mismatched coordinator metadata. Is that not true, and it's only when heights are equal?

Reply:
I could be wrong here but I when the current signer is behind, I don't send a Nack. I only do so when the current signer is more updated than what was stamped by other signers....Assuming that when this signer sends a packet with its view, other signers will eventually send a Nack with their more updated view.
I need to add an else if case for when the consensus_hash is not the same but the heights are the same....I am guessing that could mean there's a fork.